lib/systems/supported.nix

```diff
@@ -3,9 +3,14 @@
 # https://github.com/NixOS/rfcs/blob/master/rfcs/0046-platform-support-tiers.md
 { lib }:
 rec {
-  # List of systems that are built by Hydra.
-  hydra = tier1 ++ tier2 ++ tier3 ++ [
+  # List of systems that have a pure stdenv for native compilation available (in
+  # contrast to “native” stdenvs that use impure libraries like x86_64-freebsd).
+  # Musl based stdenvs are currently missing from this list.
+  pureStdenvAvailable = tier1 ++ tier2 ++ tier3 ++ [
     "aarch64-darwin"
```

Comment:
are you sure that aarch64-darwin is pure? I know the sandbox is disabled on Darwin because of some system framework restrictions but not if that's also a stdenv issue.

Reply:
Well at least we have bootstrap-tools there and build e.g. `libSystem` purely, but I think some things are impure, yes.

Reply:
I guess  we could call it `bootstrapToolsAvailable` if we want to be accurate?

Reply:
I'm sorry for the bikeshedding on this simple thing.

Given that the list of systems that the flake is using is not externally-extensible, I think it makes sense to include any system that has a chance to work. If the user is on some esoteric platform and runs `nix run nixpkgs#something`, as long as there is a `something` that would work, even after a lot of recompilation, then that system should be included. This is related to but independent from whenever the stdenv is pure or not?

Reply:
Ping me on Matrix and we can hash this out